request.js

```diff
@@ -30,6 +30,7 @@ var Tunnel = require('./lib/tunnel').Tunnel
 var now = require('performance-now')
 var Buffer = require('safe-buffer').Buffer
 
+var filterForNonReserved = helpers.filterForNonReserved
```

Comment:
I've moved this function into `helpers` to make clear the attack vector and this change's mitigation via unit tests.  If there is a better or different way to organize this, I'm all ears.